ITISFoundation/osparc-simcore
# ✨ Optional extra_credits in invitations

## What do these changes do?

This PR introduces an optional new field `extra_credits: PositiveInt`  to invitations that permits a PRODUCT_OWNER user to assign a number of credits to the newly created account's wallet upon confirmation.

### Extra details
APIs affected are:
- `services/invitations version: 1.0.2 → 1.1.0`: see  [openapi.json specs](https://editor.swagger.io/?url=https://github.com/ITISFoundation/osparc-simcore/raw/3136af03046129d95605a673fad6856590be8ab9/services/invitations/openapi.json)
   -  ✨ new field `extra_credits` to define an invitation
- `services/webserver api version: 0.32.0 → 0.33.0`: see  [openapi.json specs](https://editor.swagger.io/?url=https://raw.githubusercontent.com/ITISFoundation/osparc-simcore/3136af03046129d95605a673fad6856590be8ab9/services/web/server/src/simcore_service_webserver/api/v0/openapi.yaml)
   - ✨stores invitation in `confirmation.data` column. Loads upon confirmation and passes it to the notification so the `wallets._events` can add extra credits in the wallet.
   - 🐛 sends notification also when *confirmation email* is disabled.
  

## Related issue/s

- Implements part of  #4752 


## How to test

- Acceptance-test: 
  -  PO user creates invitation with `extraCredits=10`
  - Use invitation link
  - Complete registration and confirm to activate account
  - Login account
  - Account has a wallet with 10 credits



## DevOps 
None
